src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -409,14 +416,39 @@ private boolean isDefaultExport(TypedVar symbol) {
         || symbol.getType().isFunctionType() || isTypedef(symbol.getType());
   }
 
-  // For platform externs we skip emitting .d.ts, to avoid collisions with lib.d.ts.
-  // TODO(rado): Platform is too ill-defined, and this often filters too much.
-  // Replace with a list of externs that *only* contain symbols in lib.d.ts.
-  private boolean isPlatformExtern(String name) {
-    name = name.replace("externs.zip//", "");
-    // mostly matching what is in https://github.com/google/closure-compiler/tree/master/externs.
-    return Pattern.matches("javascript/externs/[^/]*.js", name) || name.startsWith("es")
-        || name.startsWith("w3c") || name.startsWith("ie_") || name.startsWith("browser");
+  // For platform externs we skip emitting, to avoid collisions with lib.d.ts.
+  private boolean isPlatformExtern(String filePath, String symbolName) {
+    // Some symbols are not yet provided by lib.d.ts, so we have to emit them.
+    if (emittedPlatformSymbols.contains(symbolName)) return false;
+    // This matches our test setup.
+    if (filePath.startsWith("externs.zip//")) return true;
+    String fileName = Iterables.getLast(Splitter.on("/").split(filePath));
```

Comment:
This is called hundreds of times during one run, right? What about `String fileName = filePath.substring(filePath.lastIndexOf('/') + 1)`? Alternatively, `new File(filePath).getName()`?